### PR TITLE
Add CI check for demo script

### PR DESCRIPTION
**Pull request by peternied**
### Description
Added a new check to build, install, and run the security plugin with the demo configuration script.

This was built heavily borrowing from https://github.com/opensearch-project/security-dashboards-plugin/blob/main/.github/workflows/integration-test.yml

### Issues Resolved
* Resolves https://github.com/opensearch-project/security/issues/1641

### Testing
Automatically runs as part of PR and PUSH Github actions

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
Hit a failure due to version mismatch where we updated a jar version that is behind in the distribution 

```
Caused by: java.lang.IllegalStateException: jar hell!
class: com.fasterxml.jackson.dataformat.cbor.CBORConstants
jar1: /opensearch/plugins/.installing-15168128924324[135](https://github.com/peternied/security/runs/5634491143?check_suite_focus=true#step:11:135)285/jackson-dataformat-cbor-2.13.2.jar
jar2: /opensearch/lib/jackson-dataformat-cbor-2.12.6.jar
	at org.opensearch.bootstrap.JarHell.checkClass(JarHell.java:317)
	at org.opensearch.bootstrap.JarHell.checkJarHell(JarHell.java:212)
	at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:676)
	... 11 more
```

**Comment by cliu123** [comment]
Good contribution! Looking forward to this!

**Comment by peternied** [comment]
This change is blocked on https://github.com/opensearch-project/OpenSearch/pull/2544 getting merged

File: .github/workflows/plugin_install.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      uses: whatnick/wait-action@v0.1.2
+      with:
+        time: '30s'
+
+    - name: Check OpenSearch Running
+      run: curl -XGET https://localhost:9200 -u 'admin:admin' -k -v
+
+    - name: Get Docker Logs
+      if: always()
+      run: docker logs ops
```
Comment (cliu123) [inline .github/workflows/plugin_install.yml:71]: Could you please add EOF?

**Comment by cliu123** [comment]
One of the tasks failed.
```
Run curl -XGET https://localhost:[9](https://github.com/opensearch-project/security/runs/5634568266?check_suite_focus=true#step:10:9)200 -u 'admin:admin' -k -v
Note: Unnecessary use of -X or --request, GET is already inferred.

File: .github/workflows/plugin_install.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    - name: Move and rename security plugin for installation
+      run: mv build/distributions/opensearch-security-${{env.OPENSEARCH_VERSION}}.0.zip opensearch-security.zip
+
+    - name: Run OpenSearch with plugin
+      run: |
+        cat > os-ep.sh <<EOF
+        yes | opensearch-plugin install file:///docker-host/security-plugin.zip
+        chmod +x plugins/opensearch-security/tools/install_demo_configuration.sh
+        yes | plugins/opensearch-security/tools/install_demo_configuration.sh
+        echo "plugins.security.unsupported.restapi.allow_securityconfig_modification: true" >> /opensearch/config/opensearch.yml
```
Comment (cliu123) [inline .github/workflows/plugin_install.yml:44]: Why is this needed?

File: .github/workflows/plugin_install.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/${{env.OPENSEARCH_VERSION}}/latest/linux/x64/builds/opensearch/dist/opensearch-min-${{env.OPENSEARCH_VERSION}}-linux-x64.tar.gz
+        tar -xzf opensearch-*.tar.gz
+        rm -f opensearch-*.tar.gz
+
+    - name: Move and rename security plugin for installation
+      run: mv build/distributions/opensearch-security-${{env.OPENSEARCH_VERSION}}.0.zip opensearch-security.zip
+
+    - name: Run OpenSearch with plugin
+      run: |
+        cat > os-ep.sh <<EOF
```
Comment (cliu123) [inline .github/workflows/plugin_install.yml:40]: Is it necessary to build Docker image and run OpenSearch cluster with Docker? With security plugin installed and demo configuration configured, can't OpenSearch cluster start without Docker.

File: .github/workflows/plugin_install.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    - name: Sleep while OpenSearch finishes starting up
+      uses: whatnick/wait-action@v0.1.2
+      with:
+        time: '30s'
+
+    - name: Check OpenSearch Running
+      run: curl -XGET https://localhost:9200 -u 'admin:admin' -k -v
+
+    - name: Get Docker Logs
```
Comment (cliu123) [inline .github/workflows/plugin_install.yml:69]: How does this step or the previous step verify that the cluster is up?

File: .github/workflows/plugin_install.yml (hunk 3 above)
Comment (peternied) [reply .github/workflows/plugin_install.yml:40]: I followed the model we use for integration tests for the security-dashboard-plugin, if you have thoughts on how we can shift this functionality to another model I think that would be great follow up work.

File: .github/workflows/plugin_install.yml (hunk 2 above)
Comment (peternied) [reply .github/workflows/plugin_install.yml:44]: I'm not sure if it is, this is copy and pasted from the other codebase.  I'll do an experiment if this can be removed.

File: .github/workflows/plugin_install.yml (hunk 4 above)
Comment (peternied) [reply .github/workflows/plugin_install.yml:69]: The next line `if: always()` makes sure that no matter the state of the GHA it will be run, I've thoroughly tested this feature to figure out why OpenSearch wasn't starting up due to the jackson jarhell issue that is blocking this PR

Comment (peternied) [reply .github/workflows/plugin_install.yml:69]: If any one of the actions fails the workflow is considered a failure, you can see examples of this from when I was running tests, [curl failure](https://github.com/peternied/security/runs/5633800915?check_suite_focus=true), [successful run with docker logs](https://github.com/peternied/security/runs/5633980226?check_suite_focus=true).

Read more on https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions

File: .github/workflows/plugin_install.yml (hunk 2 above)
Comment (peternied) [reply .github/workflows/plugin_install.yml]: I've removed it and the tests pass, done!

File: .github/workflows/plugin_install.yml (hunk 1 above)
Comment (peternied) [reply .github/workflows/plugin_install.yml:70]: Lets add a linter for this separate from this PR

Comment (peternied) [reply .github/workflows/plugin_install.yml:70]: See https://github.com/opensearch-project/security/pull/1699

File: build.gradle [hunk 5]
```diff
@@ -56,7 +56,7 @@ repositories {
 }
 
 ext {
-    opensearch_version = System.getProperty("opensearch.version", "1.3.0-SNAPSHOT")
+    opensearch_version = System.getProperty("opensearch.version", "1.4.0-SNAPSHOT")
```
Comment (DarshitChanpura) [inline build.gradle:59]: should we replace this with the next version?

**Review by DarshitChanpura — APPROVED** [review]

File: .github/workflows/plugin_install.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
+name: Plugin Install
+
+on: [push, pull_request, workflow_dispatch]
+
+env:
+  OPENSEARCH_VERSION: "1.4.0"
```
Comment (cliu123) [inline .github/workflows/plugin_install.yml:6]: OpenSearch 1.4.0 dependency is only on `1.x` branch. How did this pass?

Comment (peternied) [reply .github/workflows/plugin_install.yml:6]: This workflow downloads the min tar.gz of opensearch, and it runs with it.  In a follow up PR I've unified these version identifiers into one place

Comment (cliu123) [reply .github/workflows/plugin_install.yml:6]: Ok. But the OpenSearch version on `main`branch shouldn't be `1.4.0` here. Will this be fixed in the follow up PR?

File: build.gradle (hunk 5 above)
Comment (cliu123) [reply build.gradle:59]: The versions are a bit diverged. On main branch, security plugin is 2.0.0.0, but OpenSearch core version is 1.3.0

File: .github/workflows/plugin_install.yml (hunk 6 above)
Comment (peternied) [reply .github/workflows/plugin_install.yml:6]: This will be updated when we can switch to the 2.0.0 builds of OpenSearch, until then we will be pointed to the 1.4.0 to get past jar hell issues - see https://github.com/opensearch-project/OpenSearch/pull/2548
